core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/HeartBeatDefaultPayloadProviderInterface.java

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.applicationinsights.internal.heartbeat;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+
+/**
+ * <h1>Interface for setting default properties</h1>
+ *
+ * <p>
+ *   This interface is used to set the default payload of a provider and defines implementation for
+ *   some helper methods to assist it.
+ *
+ *   The default concrete implementations are {@link com.microsoft.applicationinsights.internal.heartbeat.BaseDefaultHeartbeatPropertyProvider}
+ *   and {@link com.microsoft.applicationinsights.internal.heartbeat.WebAppsDefaultHeartbeatProvider}
+ * </p>
+ *
+ * @author Dhaval Doshi
+ * @since 03-30-2018
+ */
+public interface HeartBeatDefaultPayloadProviderInterface {
+
+  /**
+   * Returns the name of the heartbeat provider.
+   * @return Name of the heartbeat provider
+   */
+  String getName();
+
+  /**
+   * Tells if the input string is a reserved property.
+   * @param keyword string to test
+   * @return true if input string is reserved keyword
+   */
+  boolean isKeyWord(String keyword);
+
+  /**
+   * Returns a callable which can be executed to set the payload based on the parameters.
+   * @param disableFields List of Properties to be excluded from payload
+   * @param provider The current heartbeat provider
+   * @return Callable which can be executed to add the payload
+   */
+  Callable<Boolean> setDefaultPayload(List<String> disableFields, HeartBeatProviderInterface provider);
```

Comment:
Nitpick: I would consider renaming this method since _get**Xyz**_ and _set**Xyz**_ are generally recognized as field accessors.

Reply:
@littleaj this was my thought, but I wanted to keep API very similar to C# implementation in this case following java camel case for consistency. Let me know if you still think that we should rename.